### ISSUE TITLE
How to test agenda with mock date

### ISSUE THREAD

**Issue by biancazzurri** [agenda/agenda#1499]
Hey guys!

We have agenda integrated on our service, basically there is a job that runs every day. We used to check that the job runs by utilizing jest-mock-date lib. It used to work, i.e. the job would fire, but now it does not fire anymore. Can you hint what is the best practice here?

Thanks

**Comment by biancazzurri** [comment]
while writing this I thought that maybe the day skipping causes the job to be missed, am i correct?

**Comment by harisvsulaiman** [comment]
Can you share a code snippet of your setup

**Comment by biancazzurri** [comment]
This is the jest test:
`    let x = 0;
    const makeJob = defineJob<Data>( 'test2', async ( data ) => x = x + data.x );
    const everyDayJob = makeJob('0 15 * * *', { x: 3 });
    await everyDayJob.schedule();
    // setTomorrow();
    const nextTime = new Date();
    nextTime.setDate(nextTime.getDate() + 1);
    nextTime.setHours(15);
    nextTime.setMinutes(0);
    nextTime.setSeconds(4);
    nextTime.setMilliseconds(0);
    MockDate.set(nextTime);
    await new Promise( resolve => setTimeout( () => resolve( 1 ), 3100 ) );
    expect(x).toBe(3);`

basically i define a job, schedule it with repeateEvery and expect it to fire when i mock tomorrow. I mock with mockdate or jest-date-mock - Same result.
